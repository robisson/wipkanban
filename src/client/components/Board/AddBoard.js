//@flow
import React from "react";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import AddIcon from "@material-ui/icons/AddCircle";
import IconButton from "@material-ui/core/IconButton";

type State = {
  open: boolean,
  name: String
};

type Props = {
  addBoard: Function
};

export default class FormDialog extends React.Component<Props, State> {
  state: State = {
    open: false,
    name: ""
  };

  handleClickOpen = () => {
    this.setState({ open: true });
  };

  handleClose = () => {
    this.props.addBoard(this.state.name);
    this.setState({ open: false });
  };

  render() {
    return (
      <>
        <IconButton
          color="default"
          aria-label="Add board"
          onClick={this.handleClickOpen}
        >
          <AddIcon />
        </IconButton>
        <Dialog
          open={this.state.open}
          onClose={this.handleClose}
          aria-labelledby="form-dialog-title"
          fullWidth={true}
          maxWidth="xs"
        >
          <DialogTitle id="form-dialog-title">Create board</DialogTitle>
          <DialogContent>
            <TextField
              autoFocus
              margin="dense"
              id="name"
              label="Type name of the board"
              fullWidth
              onChange={el => this.setState({ name: el.target.value })}
            />
          </DialogContent>
          <DialogActions>
            <Button onClick={this.handleClose} color="primary">
              Cancel
            </Button>
            <Button onClick={this.handleClose} color="primary">
              Save
            </Button>
          </DialogActions>
        </Dialog>
      </>
    );
  }
}
