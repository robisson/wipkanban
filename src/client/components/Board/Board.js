// @flow
import * as React from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Navbar from "../Navbar/index";
import Column from "../Board/Column";
import MenuLeft from "../Navbar/MenuLeft";
import MenuRight from "../Navbar/MenuRight";
import UserAccountSettings from "../Navbar/Buttons/UserAccountSettingsContainer";
import AddCardButton from "../Navbar/Buttons/AddCardButton";
import DashboardIconButton from "../Navbar/Buttons/DashboardIconButton";
import TransferWithinAStationIconButton from "../Navbar/Buttons/TransferWithinAStationIconButton";
import CalendarIconButton from "../Navbar/Buttons/CalendarIconButton";
import FilterListIconButton from "../Navbar/Buttons/FilterListIconButton";
import GroupAddIconButton from "../Navbar/Buttons/GroupAddIconButton";
import HistoryIconButton from "../Navbar/Buttons/HistoryIconButton";
import SettingsIconButton from "../Navbar/Buttons/SettingsIconButton";
import ZoomOutMapIconButton from "../Navbar/Buttons/ZoomOutMapIconButton";
import Search from "../Navbar/Search";

const styles: Object = {
  root: {
    flexGrow: 1
  },
  gridContainer: {
    overflowX: "auto",
    marginTop: 68,
    height: 700
  }
};

type Props = {
  classes: Object,
  columns: Object
};

function Board(props: Props) {
  const { classes, columns } = props;

  return (
    <div>
      <Navbar>
        <MenuLeft />
        <MenuRight>
          <AddCardButton />
          <DashboardIconButton />
          <TransferWithinAStationIconButton />
          <CalendarIconButton />
          <FilterListIconButton />
          <GroupAddIconButton />
          <HistoryIconButton />
          <SettingsIconButton />
          <ZoomOutMapIconButton />
          <Search />
          <UserAccountSettings />
        </MenuRight>
      </Navbar>
      <Grid
        container
        justify="flex-start"
        wrap="nowrap"
        spacing={0}
        className={classes.gridContainer}
      >
        {columns.map(column => (
          <Column
            key={column.id}
            vColumns={column.vColumns}
            hColumns={column.hColumns}
            title={column.title}
            cards={column.cards}
          />
        ))}
      </Grid>
    </div>
  );
}

export default withStyles(styles)(Board);
