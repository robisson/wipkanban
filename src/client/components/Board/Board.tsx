import * as React from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Navbar from "../Navbar/index";
import Column from "./Column";
import MenuLeft from "../Navbar/MenuLeft";
import MenuRight from "../Navbar/MenuRight";
import UserAccountSettings from "../Navbar/Buttons/UserAccountSettingsContainer";
import AddCardButton from "../Navbar/Buttons/AddCardButton";
import DashboardIconButton from "../Navbar/Buttons/DashboardIconButton";
import TransferWithinAStationIconButton from "../Navbar/Buttons/TransferWithinAStationIconButton";
import CalendarIconButton from "../Navbar/Buttons/CalendarIconButton";
import FilterListIconButton from "../Navbar/Buttons/FilterListIconButton";
import GroupAddIconButton from "../Navbar/Buttons/GroupAddIconButton";
import HistoryIconButton from "../Navbar/Buttons/HistoryIconButton";
import SettingsIconButton from "../Navbar/Buttons/SettingsIconButton";
import ZoomOutMapIconButton from "../Navbar/Buttons/ZoomOutMapIconButton";
import Search from "../Navbar/Search";

interface IBoardProps {
  classes: any;
  columns: any;
}

const styles: any = () => ({
  root: {
    flexGrow: 1
  },
  gridContainer: {
    overflowX: "auto",
    marginTop: 68,
    height: 700
  }
});

function BoardMain(props: IBoardProps) {
  const { classes, columns } = props;

  return (
    <div>
      <Navbar>
        <MenuLeft />
        <MenuRight>
          <AddCardButton />
          <DashboardIconButton />
          <TransferWithinAStationIconButton />
          <CalendarIconButton />
          <FilterListIconButton />
          <GroupAddIconButton />
          <HistoryIconButton />
          <SettingsIconButton />
          <ZoomOutMapIconButton />
          <Search />
          <UserAccountSettings />
        </MenuRight>
      </Navbar>
      <Grid
        container
        justify="flex-start"
        wrap="nowrap"
        spacing={0}
        className={classes.gridContainer}
      >
        {columns.map((column: any) => (
          <Column
            key={column.id}
            vColumns={column.vColumns}
            hColumns={column.hColumns}
            title={column.title}
            cards={column.cards}
          />
        ))}
      </Grid>
    </div>
  );
}

export default withStyles(styles)(BoardMain);
