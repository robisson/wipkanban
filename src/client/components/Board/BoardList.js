// @flow
import * as React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import MiniBoard from "./MiniBoard";
import Tooltip from "@material-ui/core/Tooltip";
import AddIcon from "@material-ui/icons/AddCircle";
import IconButton from "@material-ui/core/IconButton";

type Props = {
  boards: Array<Object>
};

const BoardList = ({ boards }: Props) => {
  let boardsRendered = boards.map(board => (
    <MiniBoard key={board._id} {...board} />
  ));

  return (
    <div>
      <div>
        <Typography color="secondary" variant="h6" gutterBottom>
          <DeveloperBoardIcon /> My boards
          <Tooltip title="Add team">
            <IconButton color="default" aria-label="Add board">
              <AddIcon />
            </IconButton>
          </Tooltip>
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="wrap" spacing={0}>
        <Grid item lg={6}>
          {boardsRendered}
        </Grid>
      </Grid>
    </div>
  );
};

export default BoardList;
