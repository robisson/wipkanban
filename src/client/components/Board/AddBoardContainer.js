import { connect } from "react-redux";
import AddBoard from "./AddBoard";
import { addBoard as createBoard } from "../../actions/Board";

const mapDispatchToProps = dispatch => {
  return {
    addBoard(name: String) {
      dispatch(createBoard(name));
    }
  };
};

export default connect(
  null,
  mapDispatchToProps
)(AddBoard);
