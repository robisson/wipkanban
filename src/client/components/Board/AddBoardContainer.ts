import { connect } from "react-redux";
import AddBoard from "./AddBoard";
import { addBoard as createBoard } from "../../actions/Board";
import { Dispatch } from "redux";

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    addBoard(name: String) {
      dispatch(createBoard(name));
    }
  };
};

export default connect(
  null,
  mapDispatchToProps
)(AddBoard);
