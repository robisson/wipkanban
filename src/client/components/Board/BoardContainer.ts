import Board from "./Board";
import { connect } from "react-redux";
import columns from "./fakeData";

const mapStateToProps = () => ({ columns });

export default connect(mapStateToProps)(Board);
