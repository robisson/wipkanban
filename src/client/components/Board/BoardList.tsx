// @flow
import * as React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import MiniBoard from "./MiniBoard";
import Tooltip from "@material-ui/core/Tooltip";
import AddBoard from "./AddBoardContainer";

interface IBoardListProps {
  boards: Array<Object>;
}

const BoardList: React.StatelessComponent<any> = ({
  boards
}: IBoardListProps) => {
  let boardsRendered = boards.map((board: any) => (
    <Grid key={board._id} item lg={6}>
      <MiniBoard key={board._id} {...board} />
    </Grid>
  ));

  return (
    <div>
      <div>
        <Typography color="secondary" variant="h6" gutterBottom>
          <DeveloperBoardIcon /> My boards
          <Tooltip title="Add team">
            <AddBoard />
          </Tooltip>
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="wrap" spacing={8}>
        {boardsRendered}
      </Grid>
    </div>
  );
};

export default BoardList;
