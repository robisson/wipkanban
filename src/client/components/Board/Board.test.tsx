import React from "react";
import Board from "./Board";
import columns from "./fakeData";
import { createShallow } from "@material-ui/core/test-utils";
import Grid from "@material-ui/core/Grid";
import Navbar from "../Navbar/index";
import { ShallowWrapper } from "enzyme";

describe("<Board /> Component", () => {
  let shallow: any;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it("should render correctly with Navbar and a Grid", () => {
    wrapper = shallow(<Board columns={columns} />);

    expect(wrapper.type()).toEqual('div');
    expect(wrapper.find(Navbar).type()).toEqual(Navbar);
    expect(wrapper.find(Grid).type()).toEqual(Grid);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly with Navbar and a Grid, even withou columns", () => {
    wrapper = shallow(<Board columns={[]} />);

    expect(wrapper.type()).toEqual('div');
    expect(wrapper.find(Navbar).type()).toEqual(Navbar);
    expect(wrapper.find(Grid).type()).toEqual(Grid);

    expect(wrapper).toMatchSnapshot();
  });
});
