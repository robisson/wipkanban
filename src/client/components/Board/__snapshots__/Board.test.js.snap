// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> Component should render correctly with Navbar and a Grid 1`] = `
<div>
  <WithStyles(Navbar)>
    <WithStyles(MenuLeft) />
    <WithStyles(MenuRight)>
      <AddCardButton />
      <DashboardIconButton />
      <TransferWithinAStationIconButton />
      <CalendarIconButton />
      <FilterListIconButton />
      <GroupAddIconButton />
      <HistoryIconButton />
      <SettingsIconButton />
      <ZoomOutMapIconButton />
      <Search />
      <WithStyles(Connect(UserAccountSettings)) />
    </WithStyles(MenuRight)>
  </WithStyles(Navbar)>
  <WithStyles(Grid)
    className="Board-gridContainer-2"
    container={true}
    justify="flex-start"
    spacing={0}
    wrap="nowrap"
  >
    <ColumnFactory
      cards={
        Array [
          Object {
            "id": 31,
            "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
          },
        ]
      }
      hColumns={Array []}
      key="4"
      title="A fazer"
      vColumns={Array []}
    />
    <ColumnFactory
      cards={Array []}
      hColumns={
        Array [
          Object {
            "cards": Array [],
            "hColumns": Array [],
            "id": 2,
            "title": "Chamados",
            "vColumns": Array [
              Object {
                "cards": Array [],
                "hColumns": Array [],
                "id": 4,
                "title": "Parado",
                "vColumns": Array [],
              },
              Object {
                "cards": Array [
                  Object {
                    "id": 35,
                    "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                  },
                ],
                "hColumns": Array [],
                "id": 5,
                "title": "Executando",
                "vColumns": Array [],
              },
            ],
          },
          Object {
            "cards": Array [],
            "hColumns": Array [],
            "id": 3,
            "title": "Projetos",
            "vColumns": Array [
              Object {
                "cards": Array [
                  Object {
                    "id": 36,
                    "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                  },
                ],
                "hColumns": Array [],
                "id": 41,
                "title": "Parado",
                "vColumns": Array [],
              },
              Object {
                "cards": Array [],
                "hColumns": Array [],
                "id": 51,
                "title": "Executando",
                "vColumns": Array [],
              },
            ],
          },
        ]
      }
      key="1"
      title="Executando"
      vColumns={Array []}
    />
    <ColumnFactory
      cards={
        Array [
          Object {
            "id": 32,
            "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
          },
        ]
      }
      hColumns={Array []}
      key="9"
      title="Em teste"
      vColumns={Array []}
    />
    <ColumnFactory
      cards={
        Array [
          Object {
            "id": 33,
            "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
          },
          Object {
            "id": 34,
            "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
          },
        ]
      }
      hColumns={Array []}
      key="10"
      title="Concluído"
      vColumns={Array []}
    />
  </WithStyles(Grid)>
</div>
`;

exports[`<Board /> Component should render correctly with Navbar and a Grid, even withou columns 1`] = `
<div>
  <WithStyles(Navbar)>
    <WithStyles(MenuLeft) />
    <WithStyles(MenuRight)>
      <AddCardButton />
      <DashboardIconButton />
      <TransferWithinAStationIconButton />
      <CalendarIconButton />
      <FilterListIconButton />
      <GroupAddIconButton />
      <HistoryIconButton />
      <SettingsIconButton />
      <ZoomOutMapIconButton />
      <Search />
      <WithStyles(Connect(UserAccountSettings)) />
    </WithStyles(MenuRight)>
  </WithStyles(Navbar)>
  <WithStyles(Grid)
    className="Board-gridContainer-2"
    container={true}
    justify="flex-start"
    spacing={0}
    wrap="nowrap"
  />
</div>
`;
